src/java.base/share/classes/jdk/internal/lang/stable/StableValueImpl.java

```diff
@@ -118,18 +147,24 @@ public int hashCode() {
     @Override
     public boolean equals(Object obj) {
         return obj instanceof StableValueImpl<?> other &&
-                Objects.equals(valueAcquire(), other.valueAcquire());
+                Objects.equals(orElse(null), other.orElse(null));
```

Comment:
Note that this will make `StableValue.unset` equal to `StableValue[null]`.

Reply:
Thanks for spotting this. The problem has been fixed now!